From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bradley <daniel@pulumi.com>
Date: Tue, 15 Nov 2022 13:59:57 +0000
Subject: [PATCH] Revert WAF schema changes

- This causes far too many types to be generated downstream.

diff --git a/internal/service/wafv2/schemas.go b/internal/service/wafv2/schemas.go
index 01769b9a1f..6f18fdc297 100644
--- a/internal/service/wafv2/schemas.go
+++ b/internal/service/wafv2/schemas.go
@@ -56,14 +56,14 @@ func ruleGroupRootStatementSchema(level int) *schema.Schema {
 		MaxItems: 1,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
-				"and_statement":                         statementSchema(level),
+				"and_statement":                         statementSchema(level - 1),
 				"byte_match_statement":                  byteMatchStatementSchema(),
 				"geo_match_statement":                   geoMatchStatementSchema(),
 				"ip_set_reference_statement":            ipSetReferenceStatementSchema(),
 				"label_match_statement":                 labelMatchStatementSchema(),
-				"not_statement":                         statementSchema(level),
-				"or_statement":                          statementSchema(level),
-				"rate_based_statement":                  rateBasedStatementSchema(level),
+				"not_statement":                         statementSchema(level - 1),
+				"or_statement":                          statementSchema(level - 1),
+				"rate_based_statement":                  rateBasedStatementSchema(level - 1),
 				"regex_match_statement":                 regexMatchStatementSchema(),
 				"regex_pattern_set_reference_statement": regexPatternSetReferenceStatementSchema(),
 				"size_constraint_statement":             sizeConstraintSchema(),
